packages/flutter_tools/lib/src/fuchsia/fuchsia_kernel_compiler.dart

```diff
@@ -58,7 +58,6 @@ class FuchsiaKernelCompiler {
       '--filesystem-root', fsRoot,
       '--packages', '$multiRootScheme:///$relativePackagesFile',
       '--output', fs.path.join(outDir, '$appName.dil'),
-      '--no-link-platform',
```

Comment:
@rmacnak-google @alexmarkov without removing this, apps crash on startup on a missing class in dart:ui. Is this expected?

Reply:
That sounds like there's a mismatch between the platform being used to compile the app and the platform embedded in the runner's core snapshot. Is the missing class something that was recently introduced?

Reply:
1) We shouldn't pass --no-link-platform in AOT mode (kernel compiler should complain).
2) In JIT (debug) mode, the recommended way to load platform libraries is from core snapshot, and --no-link-platform will save some time and space. If app crashes when --no-link-platform is used, it means that isolate is not initialized from core snapshot, and platform libraries were loaded from kernel file. We saw similar problem with desktop embedder recently. If, for some reason, we don't want to use core snapshots then we can remove --no-link-platform, but this is less efficient compared to spawning isolates from core snapshot.

Reply:
FYI @iskakaushik

Reply:
Looks like: https://github.com/flutter/flutter/issues/44724

Reply:
Filed: https://github.com/flutter/flutter/issues/44925

Reply:
Added TODO to add back when #44925 is fixed.